Public/Src/Sandbox/Windows/DetoursServices/DetouredFunctions.cpp

```diff
@@ -1778,6 +1778,235 @@ NTSTATUS NTAPI Detoured_ZwSetInformationFile(
+        command = wstring();
+        commandArgs = wstring();
+        return;
+    }
+
+    if (fullCommandLine[0] == L'"')
+    {
+        // Find the close quote. Might not be present which means the command
+        // is the full command line minus the initial quote.
+        size_t closeQuoteIndex = fullCommandLine.find('"', 1);
+        if (closeQuoteIndex >= 1)
+        {
+            if (closeQuoteIndex == fullCommandLine.length() - 1)
+            {
+                // Quotes cover entire command line.
+                command = fullCommandLine.substr(1, fullCommandLine.length() - 2);
+                trim_inplace(command);
+                commandArgs = wstring();
+            }
+            else
+            {
+                wstring noQuoteCommand = fullCommandLine.substr(1, closeQuoteIndex - 1);
+
+                // Find the next delimiting space after the close double-quote.
+                // For example a command line like "c:\program files"\foo we need to
+                // keep \foo and cut the quotes to produce c:\program files\foo
+                size_t spaceDelimiterIndex = fullCommandLine.find(L' ', closeQuoteIndex + 1);
+                if (spaceDelimiterIndex == wstring::npos)
+                {
+                    // No space, take everything through the end of the command line.
+                    spaceDelimiterIndex = fullCommandLine.length();
+                }
+
+                command = (noQuoteCommand +
+                    fullCommandLine.substr(closeQuoteIndex + 1, spaceDelimiterIndex - closeQuoteIndex));
+                trim_inplace(command);
+                commandArgs = fullCommandLine.substr(spaceDelimiterIndex + 1);
+                trim_inplace(commandArgs);
+            }
+        }
+        else
+        {
+            // No close quote. Take everything through the end of the command line as the command.
+            command = fullCommandLine.substr(1);
+            commandArgs = wstring();
+        }
+    }
+    else
+    {
+        // No open quote, pure space delimiter.
+        size_t spaceDelimiterIndex = fullCommandLine.find(' ');
+        if (spaceDelimiterIndex == wstring::npos)
+        {
+            // No space, take everything through the end of the command line.
+            spaceDelimiterIndex = fullCommandLine.length();
+        }
+
+        command = fullCommandLine.substr(0, spaceDelimiterIndex);
+        commandArgs = fullCommandLine.substr(spaceDelimiterIndex + 1);
+        trim_inplace(commandArgs);
+    }
+}
+
+/// Table entries used below that match process names and optional argument matches
+/// to processes that should or should not be shimmed.
+typedef struct
+{
+    const wchar_t *const ProcessName;
+
+    // Optional, if present a case-sensitive string match is performed against the command arguments.
+    // When present, this is an AND operation, i.e. the ProcessName and this search must both be found.
+    const wchar_t *const ArgMatch;
+} ShimProcessMatchInfo;

```

Comment:
> ShimProcessMatchInfo; [](start = 1, length = 22)

Does this get used somewhere?

Reply:
Nope, left over from refactoring/reimplementing. Removing.

---
In reply to: [282692819](https://github.com/microsoft/BuildXL/pull/279#discussion_r282692819) [](ancestors = 282692819)